src/coreclr/ToolBox/superpmi/superpmi-shared/methodcontext.cpp

```diff
@@ -6712,28 +6718,40 @@ int MethodContext::dumpMethodIdentityInfoToBuffer(char* buff, int len, bool igno
 
         size_t minOffset = (size_t) ~0;
         size_t maxOffset = 0;
-        uint32_t totalCount = 0;
+        uint64_t totalCount = 0;
 
         if (SUCCEEDED(pgoHR))
         {
-            // Locate the range of the counter data.
+            // Locate the range of the data.
             //
             for (UINT32 i = 0; i < schemaCount; i++)
             {
-                if ((schema[i].InstrumentationKind == ICorJitInfo::PgoInstrumentationKind::BasicBlockIntCount)
-                    || (schema[i].InstrumentationKind == ICorJitInfo::PgoInstrumentationKind::EdgeIntCount))
+                size_t start = schema[i].Offset;
+                size_t end;
+                switch (schema[i].InstrumentationKind)
                 {
-                    if (schema[i].Offset < minOffset)
-                    {
-                        minOffset = schema[i].Offset;
-                    }
+                    case ICorJitInfo::PgoInstrumentationKind::BasicBlockIntCount:
+                    case ICorJitInfo::PgoInstrumentationKind::EdgeIntCount:
+                        totalCount += *(uint32_t*)(schemaData + schema[i].Offset);
+                        end = start + 4;
+                        break;
+                    case ICorJitInfo::PgoInstrumentationKind::BasicBlockLongCount:
+                    case ICorJitInfo::PgoInstrumentationKind::EdgeLongCount:
+                        totalCount += *(uint64_t*)(schemaData + schema[i].Offset);
+                        end = start + 8;
+                        break;
+                    default:
+                        continue;
+                }
```

Comment:
I have done this so far, but shouldn't we ideally also be taking other PGO data into account here? @AndyAyersMS

Reply:
Perhaps? I left these out on purpose, though we might reconsider.

The class profile data ends up interspersed with the count data, so the hash below will typically incorporate those values; the schema records themselves don't contain that much interesting content.